packages/sitecore-jss-nextjs/src/middleware/editing-render-middleware.ts

```diff
@@ -21,13 +21,23 @@ export interface EditingRenderMiddlewareConfig {
    */
   editingDataService?: EditingDataService;
   /**
-   * Function used to determine page/route to render.
+   * Function used to determine route/page URL to render.
    * This may be necessary for certain custom Next.js routing configurations.
+   * @param {string} serverUrl The root server URL e.g. 'http://localhost:3000'
    * @param {string} itemPath The Sitecore relative item path e.g. '/styleguide'
-   * @returns {string} The route to render
-   * @default itemPath
+   * @returns {string} The URL to render
+   * @default `${serverUrl}${itemPath}`
+   * @see resolveServerUrl
    */
-  resolvePageRoute?: (itemPath: string) => string;
+  resolvePageUrl?: (serverUrl: string, itemPath: string) => string;
+  /**
+   * Function used to determine the root server URL. This is used for the route/page and subsequent data API requests.
+   * By default, the host header is used, unless the VERCEL_URL environment variable is present (for Vercel hosting).
+   * @param {NextApiRequest} req The current request.
+   * @default process.env.VERCEL_URL ? `https://${process.env.VERCEL_URL}` : `http://${req.headers.host}`;
```

Comment:
Can we determine the protocol being used? Hardcoding it seems fishy.

Reply:
@anastasiya29 Agree, we needed some context/explanation here. Per our discussion, I've switched to using `VERCEL` environment variable to determine protocol and `req.headers.host` for consistency / clarity. Updated / added comments throughout to explain.